src/resolvelib/resolvers.py

```diff
@@ -418,10 +417,29 @@ def resolve(self, requirements, max_rounds):
                 return self.state
 
             # keep track of satisfied names to calculate diff after pinning
-            satisfied_names = set(self.state.criteria.keys()) - set(
-                unsatisfied_names
+            unsatisfied_names_set = set(unsatisfied_names)
+            satisfied_names = (
+                set(self.state.criteria.keys()) - unsatisfied_names_set
             )
 
+            # If backtrack causes prefer them and their parents first
+            if self.state.backtrack_causes:
+                backtrack_cause_names = set()
+                for backtrack_cause in self.state.backtrack_causes:
+                    backtrack_cause_names.add(backtrack_cause.requirement.name)
+                    if backtrack_cause.parent:
+                        backtrack_cause_names.add(backtrack_cause.parent.name)
+
+                unsatisfied_causes_names = unsatisfied_names_set & (
+                    backtrack_cause_names
+                )
+                if unsatisfied_causes_names:
+                    unsatisfied_names = list(unsatisfied_causes_names)
+
+                self._r.backtracking_on(
+                    backtrack_cause_names, unsatisfied_causes_names
+                )
+
             # Choose the most preferred unpinned criterion to try.
             name = min(unsatisfied_names, key=self._get_preference)
```

Comment:
I could put a check here to see if the length of unsatisfied names is 1 and skip calling `self._get_preference` altogether under this scenario?